First release! (#4)
nf-core/denovotranscript: docs/usage.md

An [example samplesheet](../assets/samplesheet.csv) has been provided with the pipeline.

> ## Example workflow

Reviewer: ```suggestion
## Example workflow run
```

Reviewer: maybe move this section to the end or if you want you can have it in a separate usage docs file (see for example https://nf-co.re/taxprofiler/1.1.8/docs/usage/tutorials)

An example workflow for RNA-seq data from an experiment can look like this:

1. Run the pipeline with `--qc_only` with default params to check the quality of your reads.

```bash
nextflow run nf-core/denovotranscript \
  --input samplesheet.csv \
  --outdir <OUTDIR> \
  -profile docker \
  --qc_only
```

> 2. Run the pipeline with `qc_only` and any custom params that you have decided to use based on your data.

Reviewer: ```suggestion
2. Run the pipeline with `qc_only` and any custom parameters that you have decided to use based on your data. Use `resume` to avoid unnecessarily rerunning unchanged steps
```

> ```bash

Reviewer: ````suggestion
```bash {6-8}
````
highlights the added lines

nextflow run nf-core/denovotranscript \
  --input samplesheet.csv \
  --outdir <OUTDIR> \
  -profile docker \
  --qc_only \
  --extra_fastp_args='--trim_front1 15 --trim_front2 15' \
  --remove_ribo_rna \
  -resume
```

3. Once you are happy with the quality of your reads, run the pipeline without `qc_only` to proceed to transcriptome assembly.
   If you have a lot of samples, you may wish to only include one replicate per condition in the samplesheet for assembly, as this may be quicker to run. This command also runs transcriptome quality assessment steps and quantification.

> ```bash

Reviewer: ```suggestion
```bash {2,7,8}
```

nextflow run nf-core/denovotranscript \
  --input samplesheet_assembly.csv \ # samplesheet with only one replicate per sample
  --outdir <OUTDIR> \
  -profile docker \
  --extra_fastp_args='--trim_front1 15 --trim_front2 15' \
  --remove_ribo_rna \
  --extra_tr2aacds_args='-MINAA 50' \
  --busco_lineage= 'arthropoda_odb10' \
  -resume
```

> 4. If you created the transcriptome assembly with only one replicate per condition, you can now run the pipeline with the full samplesheet, path to `--transcript_fasta`, and `--skip_assembly` to quantify the expression of all samples. If you used a samplesheet with all samples in step 3, this step is not needed.

Reviewer: something is off in this sentence after "full samplesheeet"

Author: Thanks, I've split it into two sentences.

> ```bash

Reviewer: ```suggestion
```bash {2,3,8}
```

There is also an optional step to remove rRNA reads using SortMeRNA after trimming using the `--remove_ribo_rna` parameter. You can provide a custom ribosomal RNA database with the `--ribo_database_manifest` parameter. By default, the pipeline uses the databases mentioned in `assets/rrna-db-defaults.txt`. You can also save the non-ribo reads with the `--save_non_ribo_reads` parameter. If
rRNA removal is performed, then FastQC is run again on the non-ribo reads.

## Assembly and redundancy reduction options

The `--assemblers` parameter can be used to specify which assemblers to use. By default, the pipeline uses Trinity and rnaSPAdes (`--assemblers "trinity,rnaspades"`).
The third assembler option you can add is `"trinity_no_norm"` for running Trinity without normalised reads. Extra parameters can be provided to Trinity using the `extra_trinity_args` parameter.

For rnaSPAdes, the following additional params can be used:

- `--soft_filtered_transcripts` (to include soft filtered transcripts from rnaSPAdes as inputs to Evidential Gene)
- `--hard_filtered_transcripts` (to include hard filtered transcripts from rnaSPAdes as inputs to Evidential Gene)

> - `--ss` param can be used to set the strand-specific type for rnaSPAdes.

Reviewer: ```suggestion
- `--strand_specific` parameter can be used to set the strand-specific type for rnaSPAdes.
```
or is that parameter name coming from rnaSPAdes?

Author: This param is coming from rnaSPAdes, so I will leave it as `ss`
